sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedParquetRecordReader.java

```diff
@@ -298,7 +298,7 @@ private void initializeInternal() throws IOException, UnsupportedOperationExcept
 
   private void checkEndOfRowGroup() throws IOException {
     if (rowsReturned != totalCountLoadedSoFar) return;
-    PageReadStore pages = reader.readNextRowGroup();
+    PageReadStore pages = reader.readNextFilteredRowGroup();
```

Comment:
To benchmark [Column indexes](https://issues.apache.org/jira/browse/PARQUET-1201).

Reply:
It seems slower than Parquet 1.10.1. cc @gszadovszky 

```shell
SPARK_GENERATE_BENCHMARK_FILES=1 build/sbt "sql/test:runMain org.apache.spark.sql.execution.benchmark.ParquetFilterPushdownBenchmark"
```

Case | Parquet 1.11 Vectorized(ms) | Parquet 1.11 Vectorized(Pushdown)(ms) | Parquet 1.10 Vectorized(ms) | Parquet 1.10 Vectorized(Pushdown)(ms)
-- | -- | -- | -- | --
Case 1 | 9459 | 604 | 8378 | 487
Case 2 | 10394 | 588 | 8332 | 496
Case 3 | 12594 | 980 | 9201 | 495
Case 4 | 15398 | 1288 | 9890 | 997
Case 5 | 11817 | 568 | 10780 | 1284
Case 6 | 20275 | 17979 | 17476 | 18447
Case 7 | 6985 | 486 | 7211 | 469
Case 8 | 6983 | 482 | 7062 | 448
Case 9 | 7002 | 481 | 6990 | 441
Case 10 | 7041 | 488 | 6983 | 451
Case 11 | 7050 | 490 | 6978 | 440
Case 12 | 6973 | 474 | 6958 | 441
Case 13 | 7620 | 1852 | 7691 | 1759
Case 14 | 10130 | 6955 | 9942 | 6707
Case 15 | 12529 | 12156 | 12268 | 11739
Case 16 | 13067 | 13478 | 12762 | 12810
Case 17 | 13031 | 13520 | 12740 | 12873
Case 18 | 13497 | 13866 | 12704 | 13002
Case 19 | 6878 | 422 | 6850 | 397
Case 20 | 6861 | 416 | 6855 | 384
Case 21 | 6931 | 574 | 7225 | 560
Case 22 | 6897 | 578 | 6945 | 543
Case 23 | 6924 | 571 | 6944 | 540
Case 24 | 14593 | 15393 | 14458 | 14481
Case 25 | 10076 | 2612 | 9167 | 2384
Case 26 | 7761 | 489 | 7889 | 462
Case 27 | 7578 | 475 | 7840 | 450
Case 28 | 3608 | 125 | 3602 | 119
Case 29 | 4888 | 2118 | 4920 | 2137
Case 30 | 9168 | 9046 | 9280 | 8844
Case 31 | 10327 | 10353 | 10183 | 10177
Case 32 | 3775 | 125 | 3741 | 118
Case 33 | 4431 | 1159 | 4362 | 1141
Case 34 | 7096 | 5558 | 7046 | 5258
Case 35 | 10186 | 9490 | 9769 | 9356
Case 36 | 5218 | 135 | 5367 | 135
Case 37 | 6092 | 1517 | 6145 | 1489
Case 38 | 9531 | 7077 | 9565 | 7032
Case 39 | 13191 | 12561 | 12879 | 12399
Case 40 | 7652 | 511 | 6870 | 457
Case 41 | 6973 | 500 | 6844 | 457
Case 42 | 6992 | 499 | 7015 | 466
Case 43 | 7017 | 511 | 7148 | 529
Case 44 | 7059 | 519 | 7011 | 494
Case 45 | 6999 | 518 | 6886 | 485
Case 46 | 7262 | 7408 | 7134 | 7287
Case 47 | 7221 | 7346 | 7123 | 7282
Case 48 | 7190 | 7388 | 7246 | 7262
Case 49 | 7144 | 7358 | 7093 | 7286
Case 50 | 7725 | 7388 | 7035 | 7241
Case 51 | 7143 | 7316 | 7104 | 7281
Case 52 | 3946 | 234 | 4075 | 228
Case 53 | 4486 | 1187 | 4550 | 1164
Case 54 | 7094 | 5187 | 6915 | 5105
Case 55 | 9375 | 9067 | 9791 | 9074
Case 56 | 4186 | 4216 | 4103 | 4210
Case 57 | 4815 | 4867 | 4738 | 4730
Case 58 | 7500 | 7505 | 7175 | 7701
Case 59 | 9836 | 9885 | 9727 | 9704
Case 60 | 3694 | 125 | 3787 | 116
Case 61 | 4351 | 1185 | 4252 | 1090
Case 62 | 7279 | 5168 | 6708 | 5011
Case 63 | 9353 | 9033 | 9633 | 8888
Case 64 | 3850 | 126 | 3812 | 116
Case 65 | 4531 | 1141 | 4464 | 1105
Case 66 | 7053 | 5072 | 6881 | 5078
Case 67 | 9171 | 8913 | 9896 | 8989
Case 68 | 201 | 204 | 209 | 211
Case 69 | 835 | 904 | 843 | 916
Case 70 | 2577 | 2888 | 2609 | 2934
Sum | 544365 | 291698 | 524055 | 285089